### DISCUSSION TITLE
Can not write to SMB/CIFS mounted Docker volume

### DISCUSSION THREAD

**Discussion by noveb** [jlesage/docker-handbrake#284] [Q&A]
Hi everyone,

I have a setup with the handbrake container on one VM and a SMB/CIFS server on another VM. I created 2 CIFS volumes in Docker and mount them to /storage and /output. Sadly I get this error
`[autovideoconverter] ERROR: Cannot process watch folder '/watch', because the associated output directory '/output' is not writable.`

I already tried some things: if I make the volume read-only, this will fail on startup
`[cont-init   ] 55-handbrake.sh: executing...
[cont-init   ] 55-handbrake.sh: core dump file location: core
[cont-init   ] 55-handbrake.sh: core dump file size: unlimited (blocks)
[cont-init   ] 55-handbrake.sh: terminated successfully.`

I can also exec into the handbrake container and do `touch myfile` and `mkdir mydir` in /output without problems.

Does anyone have an idea what I am missing and what the problem is?
Greetings

**Comment by jlesage** [comment]
How did you create the container and how did you create the CIFS volumes ?  There is probably a mismatch between the user of the container (specified via `USER_ID` and `GROUP_ID`) and the user of the mounted CIFS volumes.

**Reply by noveb** [reply]
I use Portainer as UI but basically i did this https://docs.docker.com/storage/volumes/#create-cifssamba-volumes

When i exec into the container, I am the root user and can write to the volume. So I guess there is then something up with permissions? I thought this kind of volume would hide the underlying user & auth stuff from the container. User ID on the smb server is 1004 but that does not even exist on the docker host. I am a bit clueless now :D

**Reply by jlesage** [reply]
Inside the container, you can do `ls -l /storage` to see user/group associate to the files/folders.  You should probably re-use the same.
It's not a problem if this user/group doesn't exists on the Docker host.
